src/mqtt.cpp

```diff
@@ -255,12 +258,42 @@ namespace mqtt {
     }
   }
 
+  void sendStatus(const char *newmsg) {
+    MqttMessage msg;
+    msg.cmd = X_CMD_PUBLISH_STATUSMSG;
+    sprintf(statusmsg, "%s", newmsg);
```

Comment:
There's a rare chance that a sendStatusInteral call is in progress while a sentStatus call is made from another thread which then overwrites/corrupts the `statusmsg`. Can we avoid that e.g. by passing the status string (or a constant that gets resolved to a string) with the message? How long do you anticipate the statue message to get?

Reply:
Not sure if there is a much better solution, and if its worth increasing the queue message size which would reserve a larger chunk of memory. If we could keep the status message short then adding it to the queue message could work, otherwise not so sure.

Reply:
I originally was storing the string in the message, but couldn't make it work through the queue, so fell back to the global object, but you're right it's messy... I took another look and found the  bug that was breaking the queue passing (wrong size being passed to the createQueue method).

Reply:
Just wondering - wouldn't the same approach of sending a pointer to the JsonDocument have worked for a String pointer as well as long as the publishStatusInternal method deleted it to free the memory?
I do like the idea of having an option to publish structured data, but it feels a little over kill for publishing a simple status message. What do you think, and do you have future changes that would make use of a generic publish for structured data